cmd/discord-provider/server/operations.go

```diff
@@ -16,8 +22,46 @@ func NewServer() *Server {
 }
 
 func (server *Server) SendMessage(ctx context.Context, message *discord_provider.SendMessageRequest) (*discord_provider.SendMessageResponse, error) {
-	// Sends a message to a Discord webhook here
+	url, found := os.LookupEnv("DISCORD_WEBHOOK_URL")
+	if !found {
+		return &discord_provider.SendMessageResponse{}, errors.New("the env var DISCORD_WEBHOOK_URL is not set to a value")
+	}
+
+	discordMessage := discordMessage{
+		Username: "calidum-rotae",
+		Embeddeds: []embedded{
+			{
+				Title: "New submission",
+				Description: discordSenderInformation{
+					firstName:      message.Sender.FirstName,
+					lastName:       message.Sender.LastName,
+					email:          message.Sender.Email,
+					phoneNumber:    message.Sender.PhoneNumber,
+					requestDetails: message.RequestDetails,
+					requestService: message.RequestService,
+				}.String(),
+				Color: "16745728", // orange
+				Footer: footer{
+					Text: "By calidum-rotae services",
+				},
+			}},
+	}
+
+	payload := new(bytes.Buffer)
+	err := json.NewEncoder(payload).Encode(discordMessage)
+	if err != nil {
+		return &discord_provider.SendMessageResponse{}, fmt.Errorf("failed to encode the message %v", discordMessage)
+	}
+
+	resp, err := http.Post(url, "application/json", payload)
+	if err != nil {
+		defer resp.Body.Close()
+		respBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return &discord_provider.SendMessageResponse{}, fmt.Errorf("failed to send discord webhook: status code: %d\n body: %s", resp.StatusCode, string(respBody))
```

Comment:
This line and line 63 should be swapped, i think. Otherwise this'll crash, since the http body variable will be equal to "nil" and thus the formatting will print an empty variable. If this line is swapped with line 63, the http body will be printed.

Reply:
Yup, nice catch. I will make the change